gossipd/routing.c

```diff
@@ -1187,6 +1195,8 @@ u8 *handle_node_announcement(struct routing_state *rstate, const u8 *node_ann)
 	/* Beyond this point it's not malformed, so safe if we make it
 	 * pending and requeue later. */
 	node = get_node(rstate, &node_id);
+	if (!node_has_public_channels(node))
```

Comment:
`node_has_public` dereferences `node` which may be `NULL`, this needs a check.

Reply:
That will teach me: I originally rewrote the code, then decided it was clearer as two patches: the  minimal fix, then the rework.  So I hand-hacked in the minimal version.

Fixed!